ring/lifecycler.go

```diff
@@ -759,11 +779,43 @@ func (i *Lifecycler) compareTokens(fromRing Tokens) bool {
 	return true
 }
 
+func (i *Lifecycler) canJoin(ctx context.Context) error {
+	desc, err := i.KVStore.Get(ctx, i.RingKey)
```

Comment:
This needs to be called before each call to `CanJoin()`. Otherwide `desc` will never change and then how can the state change when you call `CanJoin()` in the `for` loop?

Reply:
Hi @pracucci,
This is probably due to my misunderstanding of the result returned by `i.KVStore.Get()`. My interpretation was:
- `desc` of type `*Desc` is returned by `i.KVStore.Get()`
- inside the loop we call `desc.GetIngesters()`, which gives us the latest map of ingesters
- during the waiting times, somebody calls `desc.AddIngester(newIngesterDesc)` and this way the next call to `desc.GetIngesters()` will include `newIngesterDesc` as well.

I would like to understand what is wrong with this interpretation. Anyway, I moved the call to `i.KVStore.Get()` inside the loop.